File: GXpert/GXpert.Web/Modules/Activation/Activation/ActivationForm.cs
﻿using Serenity.ComponentModel;
using System;

namespace GXpert.Activation.Forms;

[FormScript("Activation.Activation")]
[BasedOnRow(typeof(ActivationRow), CheckNames = true)]
public class ActivationForm
{
    public int PlayListId { get; set; }
    public int TeacherId { get; set; }
    public int ActivationLogId { get; set; }
    public string DeviceId { get; set; }
    public string DeviceDetails { get; set; }
    public DateTime ActivationDate { get; set; }
    public DateTime ExpiryDate { get; set; }
    public short EStatus { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public short IsActive { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/ActivationForm.ts
﻿import { IntegerEditor, ServiceLookupEditor, StringEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ActivationForm {
    PlayListId: IntegerEditor;
    TeacherId: IntegerEditor;
    ActivationLogId: ServiceLookupEditor;
    DeviceId: StringEditor;
    DeviceDetails: StringEditor;
    ActivationDate: DateEditor;
    ExpiryDate: DateEditor;
    EStatus: IntegerEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
    IsActive: IntegerEditor;
}

export class ActivationForm extends PrefixedContext {
    static readonly formKey = 'Activation.Activation';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ActivationForm.init)  {
            ActivationForm.init = true;

            var w0 = IntegerEditor;
            var w1 = ServiceLookupEditor;
            var w2 = StringEditor;
            var w3 = DateEditor;

            initFormType(ActivationForm, [
                'PlayListId', w0,
                'TeacherId', w0,
                'ActivationLogId', w1,
                'DeviceId', w2,
                'DeviceDetails', w2,
                'ActivationDate', w3,
                'ExpiryDate', w3,
                'EStatus', w0,
                'InsertDate', w3,
                'InsertUserId', w0,
                'UpdateDate', w3,
                'UpdateUserId', w0,
                'IsActive', w0
            ]);
        }
    }
}